pulsar-client-cpp/lib/ProducerConfiguration.cc

```diff
@@ -211,7 +211,7 @@ ProducerConfiguration& ProducerConfiguration::setLazyStartPartitionedProducers(
 }
 
 bool ProducerConfiguration::getLazyStartPartitionedProducers() const {
-    return impl_->useLazyStartPartitionedProducers;
+    return impl_->useLazyStartPartitionedProducers && impl_->accessMode == Shared;
```

Comment:
It's a trivial getter, we should not check other fields. Instead, we should call these two getters in `PartitionedProducerImpl` like https://github.com/apache/pulsar/blob/2848fa0da09e035951220c3d04138041e1477e60/pulsar-client/src/main/java/org/apache/pulsar/client/impl/PartitionedProducerImpl.java#L91-L92